### PR TITLE
SDK-373: php task set doesn't work. 

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
[SDK-373](https://spryker.atlassian.net/browse/SDK-373)

### CODE REVIEW

File: extension/Hello/Tasks/ByeWorldTask.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+id: 'bye:world'
+short_description: 'Bye bye'
+help: 'Will greet the one using it'
+stage: bye
+version: 1.0.0
+deprecated: false
+successor: 'hello:php'
+command: '/bin/echo "bye %world%"'
+type: local_cli
+tags: ['bye', 'world']
```
Comment (vol4onok) [inline extension/Hello/Tasks/ByeWorldTask.yaml:11]: we already have tags in task set. in this case if we provide tags from task set we need overwrite these ones for task set

Comment (Kyrylo-Meretskyi) [reply extension/Hello/Tasks/ByeWorldTask.yaml:11]: Got it, I'll implement overriding of tags

File: src/Core/Appplication/Service/TaskExecutor.php [hunk 2]
```diff
@@ -155,33 +187,42 @@ protected function executeTasks(ContextInterface $context): ContextInterface
...
+        ) {
+            return $context;
+        }
+
+        $stageCommands = array_filter($task->getCommands(), function (CommandInterface $command) use ($stage): bool {
+            return $command->getStage() === $stage;
         });
 
-        if (count($stageTasks) > 0) {
+        if ($context->getTags()) {
```
Comment (vol4onok) [inline src/Core/Appplication/Service/TaskExecutor.php:206]: remove tags from context. it's located in command

File: src/Infrastructure/Repository/TaskYamlRepository.php [hunk 3]
```diff
@@ -406,26 +406,55 @@ protected function buildTask(array $taskData, array $taskListData, array $tags =
...
+            $subTask = $tasks[$subTaskData['id']] ?? null;
+
+            if ($subTask === null) {
                 continue;
             }
 
-            $taggedTask->setTags($taggedTaskData['tags'] ?? []);
+            $subTaskCommands = $this->addStageToSubTasks($subTask, $subTask->getCommands());
+
+            $taskSetCommands = array_merge($subTaskCommands, $taskSetCommands);
```
Comment (vol4onok) [inline src/Infrastructure/Repository/TaskYamlRepository.php:431]: move array_merge from foreach. you can use it like
$taskSetCommands[] = subTaskCommands; in foreach and then

$taskSetCommands = array_merge(...$taskSetCommands)

File: src/Presentation/Console/Commands/TaskRunFactoryLoader.php [hunk 4]
```diff
@@ -186,16 +185,8 @@ protected function addTagOptions(TaskInterface $task, array $options): array
...
-        }
-
-        if ($task instanceof TaskSetInterface) {
-            foreach ($task->getSubTasks() as $taskSetTask) {
-                if ($taskSetTask instanceof TaggedTaskInterface) {
-                    $tags = array_merge($tags, $taskSetTask->getTags());
-                }
-            }
+        foreach ($task->getCommands() as $command) {
+            $tags = array_merge($tags, $command->getTags());
```
Comment (vol4onok) [inline src/Presentation/Console/Commands/TaskRunFactoryLoader.php:189]: the same here

**Review by vol4onok — APPROVED** [review]

File: src/Presentation/Console/Commands/TaskRunFactoryLoader.php (hunk 4 above)
Comment (Kyrylo-Meretskyi) [reply src/Presentation/Console/Commands/TaskRunFactoryLoader.php]: Fixed

**Review by vol4onok — APPROVED** [review]

**Pull request merged**